nemo/collections/asr/audio_preprocessing.py

```diff
@@ -738,6 +742,163 @@ def output_ports(self):
         }
 
 
+class TimeStretchAugmentation(NonTrainableNM):
+    def __init__(
+        self,
+        sample_rate: int,
+        probability: float,
+        min_speed_rate: float = 0.9,
+        max_speed_rate: float = 1.1,
+        num_rates: int = 5,
+        n_fft: int = 512,
+    ):
+        """
+        Time-stretch an audio series by a fixed rate while preserving pitch.
+
+        Note:
+        This is a simplified implementation, intended primarily for reference and pedagogical purposes.
+        It makes no attempt to handle transients, and is likely to produce audible artifacts.
+
+        Args:
+            sample_rate: Sampling rate.
+            probability: Float value declaring chance of the input being augmented.
+                Must be a float value in the range [0, 1].
+            min_speed_rate: Minimum sampling rate modifier.
+            max_speed_rate: Maximum sampling rate modifier.
+            num_rates: Number of discrete rates to allow. Can be a positive or negative
+                integer.
+                If a positive integer greater than 0 is provided, the range of
+                speed rates will be discretized into `num_rates` values.
+                If a negative integer or 0 is provided, the full range of speed rates
+                will be sampled uniformly.
+                Note: If a positive integer is provided and the resultant discretized
+                range of rates contains the value '1.0', then those samples with rate=1.0,
+                will not be augmented at all and simply skipped. This is to avoid unnecessary
+                augmentation and increase computation time. Effective augmentation chance
+                in such a case is = `prob * (num_rates - 1 / num_rates) * 100`% chance
+                where `prob` is the global probability of a sample being augmented.
+            n_fft: Number of fft filters to be computed.
+        """
+        super(TimeStretchAugmentation, self).__init__()
+
+        if probability > 1.0 or probability < 0.0:
+            raise ValueError("`probability` must be between 0 and 1")
+
+        if not HAVE_TORCHAUDIO:
+            raise ModuleNotFoundError(
+                "torchaudio is not installed but is necessary for "
+                "TimeStretchAugmentation. We recommend you try "
+                "installing it from conda for the PyTorch version you have."
+            )
```

Comment:
Let's add torchaudio to nemo_asr requirements, and remove these checks.

Reply:
Sounds good.

Reply:
Looking at torchaudio's installation procedure, it seems they recommend conda installs as well. The pip install version (https://github.com/pytorch/audio) fails on Windows, but the conda install works. I think its best to not explicitly add the pip version to requirements_asr.txt so as not to break windows installations.

Reply:
I wouldn't be too concerned about windows, but can you raise this issue on slack?

Reply:
Hmm, even for linux and macosx, torchaudio has the `--find-links https://download.pytorch.org/whl/torch_stable.html` line in its installation, which has links to compiled binaries. That line has to be called at pip level, and cant be added into the requirements_asr.txt file itself. 

In my opinion, it is probably best to inform users that certain features will be disabled unless they install torchaudio separately as is currently done for certain pre-processing steps like MFCC.